pkg/watch/errorlist_test.go

```diff
@@ -0,0 +1,75 @@
+package watch
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func Test_WatchesError(t *testing.T) {
+	someErr := errors.New("some err")
+	someGVKA := schema.GroupVersionKind{Group: "a", Version: "b", Kind: "c"}
+	someGVKB := schema.GroupVersionKind{Group: "x", Version: "y", Kind: "z"}
+
+	type errsToAdd struct {
+		errs    []error
+		gvkErrs []gvkErr
+	}
+
+	for _, tt := range []struct {
+		name         string
+		errsToAdd    errsToAdd
+		expectedGVKs []schema.GroupVersionKind
+		generalErr   bool
+	}{
+		{
+			name: "gvk errors, no global",
+			errsToAdd: errsToAdd{
+				gvkErrs: []gvkErr{
+					{err: someErr, gvk: someGVKA},
+					{err: someErr, gvk: someGVKB},
+				},
+			},
+			expectedGVKs: []schema.GroupVersionKind{someGVKA, someGVKB},
+			generalErr:   false,
+		},
+		{
+			name: "gvk errors and global",
+			errsToAdd: errsToAdd{
+				gvkErrs: []gvkErr{
+					{err: someErr, gvk: someGVKA},
+					{err: someErr, gvk: someGVKB},
+				},
+				errs: []error{someErr, someErr},
+			},
+			expectedGVKs: []schema.GroupVersionKind{someGVKA, someGVKB},
+			generalErr:   true,
+		},
+		{
+			name: "just global",
+			errsToAdd: errsToAdd{
+				gvkErrs: []gvkErr{},
```

Comment:
removing this line will give a nil slice that evaluates to the same thing in the loop